File: frontend/src/interface/forklift.ts
export type ForkliftInterface = {
  id: number;
  brand: string;
  number: string;
  loadCapacity: string;
  isActive: boolean;
  updateDate: string;
  fio: string;
};
export type ForkliftCreateInterface = Omit<
  ForkliftInterface,
  "id" | "isActive" | "updateDate" | "fio"
>;

export type ForkliftUpdateInterface = Omit<
  ForkliftInterface,
  "isActive" | "updateDate" | "fio"
>;

export type SearchFormData = {
  forkliftNumber: string;
};

File: backend/src/main/java/by/alexander/backend/entity/Forklift.java
package by.alexander.backend.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;
import lombok.experimental.FieldDefaults;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "forklifts")
@Data
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Accessors(chain = true)
public class Forklift {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    Long id;

    @Column(name = "brand")
    String brand;

    @Column(name = "number")
    String number;

    @Column(name = "load_capacity")
    BigDecimal loadCapacity;

    @Column(name = "is_active")
    Boolean isActive;

    @Column(name = "update_date")
    LocalDateTime updateDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    User user;
}
